src/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs

```diff
@@ -276,19 +276,17 @@ public string UserName
 
                 int userNameLen = 256;
                 int domainNameLen = 256;
-                int sidNameUse = 0;
-                StringBuilder bufUserName = new StringBuilder(userNameLen);
-                StringBuilder bufDomainName = new StringBuilder(domainNameLen);
-                StringBuilder retUserName = new StringBuilder();
-
-                if (Interop.Kernel32.LookupAccountSid(MachineName, sid, bufUserName, ref userNameLen, bufDomainName, ref domainNameLen, ref sidNameUse) != 0)
+                unsafe
                 {
-                    retUserName.Append(bufDomainName.ToString());
-                    retUserName.Append("\\");
-                    retUserName.Append(bufUserName.ToString());
+                    char* bufUserName = stackalloc char[userNameLen];
+                    char* bufDomainName = stackalloc char[domainNameLen];
```

Comment:
allocating 1K on the stack is not too much?

Reply:
We generally try to stay under it, but it seemed reasonable here.  I could allocate though; it'd be no worse than it was.